src/components/dialog.js

```diff
@@ -0,0 +1,102 @@
+(() => ({
+  name: 'Dialog',
+  type: 'CONTAINER_COMPONENT',
+  allowedTypes: ['BODY_COMPONENT', 'CONTENT_COMPONENT'],
+  orientation: 'HORIZONTAL',
+  jsx: (() => {
+    const { env } = B;
+    const { Dialog } = window.MaterialUI.Core;
+
+    const isEmpty = children.length === 0;
+    const isDev = env === 'dev';
+    const isPristine = isDev && isEmpty;
+    const isContentVisible = isDev && options.isVisibleInDev;
+
+    const classNames = isContentVisible
+      ? [
+          classes.content,
+          isEmpty ? classes.empty : '',
+          isPristine ? classes.pristine : '',
+        ].join(' ')
+      : '';
+
+    const [open, setOpen] = React.useState(false);
+    const devContent = isContentVisible && !isEmpty ? children : 'Dialog';
+
+    useEffect(() => {
+      B.defineFunction('handleDialogOpen', e => {
+        e.preventDefault();
+
+        setOpen(true);
+      });
+
+      B.defineFunction('handleDialogClose', e => {
+        e.preventDefault();
+
+        setOpen(false);
+      });
+    }, []);
+
+    return (
+      <div className={classNames}>
+        {isDev ? devContent : ''}
+
+        <Dialog open={open} onClose={() => setOpen(false)}>
+          <div className={classes.dialog}>{children}</div>
```

Comment:
Why are you using `<div className={classes.dialog}>` to show the children?
Material UI offers `<DialogTitle>`, `<DialogContent>`, `<DialogContentText>`, `<DialogActions>` to show content in. Better use those instead.

Reply:
I'd like to add the options in a next iteration, since the dialog is functional as is and you are able to add title and other content just by dragging and dropping them into the dialog. It seems like a good solution. The options that Material UI offers might make it more simple for the end user, but that sounds like a nice to have for a later iteration.